// @dank074/discord-video-stream/dist/client/GatewayOpCodes.js
export var GatewayOpCodes;
(function (GatewayOpCodes) {
    GatewayOpCodes[GatewayOpCodes["DISPATCH"] = 0] = "DISPATCH";
    GatewayOpCodes[GatewayOpCodes["HEARTBEAT"] = 1] = "HEARTBEAT";
    GatewayOpCodes[GatewayOpCodes["IDENTIFY"] = 2] = "IDENTIFY";
    GatewayOpCodes[GatewayOpCodes["PRESENCE_UPDATE"] = 3] = "PRESENCE_UPDATE";
    GatewayOpCodes[GatewayOpCodes["VOICE_STATE_UPDATE"] = 4] = "VOICE_STATE_UPDATE";
    GatewayOpCodes[GatewayOpCodes["VOICE_SERVER_PING"] = 5] = "VOICE_SERVER_PING";
    GatewayOpCodes[GatewayOpCodes["RESUME"] = 6] = "RESUME";
    GatewayOpCodes[GatewayOpCodes["RECONNECT"] = 7] = "RECONNECT";
    GatewayOpCodes[GatewayOpCodes["REQUEST_GUILD_MEMBERS"] = 8] = "REQUEST_GUILD_MEMBERS";
    GatewayOpCodes[GatewayOpCodes["INVALID_SESSION"] = 9] = "INVALID_SESSION";
    GatewayOpCodes[GatewayOpCodes["HELLO"] = 10] = "HELLO";
    GatewayOpCodes[GatewayOpCodes["HEARTBEAT_ACK"] = 11] = "HEARTBEAT_ACK";
    GatewayOpCodes[GatewayOpCodes["CALL_CONNECT"] = 13] = "CALL_CONNECT";
    GatewayOpCodes[GatewayOpCodes["GUILD_SUBSCRIPTIONS"] = 14] = "GUILD_SUBSCRIPTIONS";
    GatewayOpCodes[GatewayOpCodes["LOBBY_CONNECT"] = 15] = "LOBBY_CONNECT";
    GatewayOpCodes[GatewayOpCodes["LOBBY_DISCONNECT"] = 16] = "LOBBY_DISCONNECT";
    GatewayOpCodes[GatewayOpCodes["LOBBY_VOICE_STATES_UPDATE"] = 17] = "LOBBY_VOICE_STATES_UPDATE";
    GatewayOpCodes[GatewayOpCodes["STREAM_CREATE"] = 18] = "STREAM_CREATE";
    GatewayOpCodes[GatewayOpCodes["STREAM_DELETE"] = 19] = "STREAM_DELETE";
    GatewayOpCodes[GatewayOpCodes["STREAM_WATCH"] = 20] = "STREAM_WATCH";
    GatewayOpCodes[GatewayOpCodes["STREAM_PING"] = 21] = "STREAM_PING";
    GatewayOpCodes[GatewayOpCodes["STREAM_SET_PAUSED"] = 22] = "STREAM_SET_PAUSED";
    GatewayOpCodes[GatewayOpCodes["REQUEST_GUILD_APPLICATION_COMMANDS"] = 24] = "REQUEST_GUILD_APPLICATION_COMMANDS";
    GatewayOpCodes[GatewayOpCodes["EMBEDDED_ACTIVITY_LAUNCH"] = 25] = "EMBEDDED_ACTIVITY_LAUNCH";
    GatewayOpCodes[GatewayOpCodes["EMBEDDED_ACTIVITY_CLOSE"] = 26] = "EMBEDDED_ACTIVITY_CLOSE";
    GatewayOpCodes[GatewayOpCodes["EMBEDDED_ACTIVITY_UPDATE"] = 27] = "EMBEDDED_ACTIVITY_UPDATE";
    GatewayOpCodes[GatewayOpCodes["REQUEST_FORUM_UNREADS"] = 28] = "REQUEST_FORUM_UNREADS";
    GatewayOpCodes[GatewayOpCodes["REMOTE_COMMAND"] = 29] = "REMOTE_COMMAND";
    GatewayOpCodes[GatewayOpCodes["GET_DELETED_ENTITY_IDS_NOT_MATCHING_HASH"] = 30] = "GET_DELETED_ENTITY_IDS_NOT_MATCHING_HASH";
    GatewayOpCodes[GatewayOpCodes["REQUEST_SOUNDBOARD_SOUNDS"] = 31] = "REQUEST_SOUNDBOARD_SOUNDS";
    GatewayOpCodes[GatewayOpCodes["SPEED_TEST_CREATE"] = 32] = "SPEED_TEST_CREATE";
    GatewayOpCodes[GatewayOpCodes["SPEED_TEST_DELETE"] = 33] = "SPEED_TEST_DELETE";
    GatewayOpCodes[GatewayOpCodes["REQUEST_LAST_MESSAGES"] = 34] = "REQUEST_LAST_MESSAGES";
    GatewayOpCodes[GatewayOpCodes["SEARCH_RECENT_MEMBERS"] = 35] = "SEARCH_RECENT_MEMBERS";
    GatewayOpCodes[GatewayOpCodes["REQUEST_CHANNEL_STATUSES"] = 36] = "REQUEST_CHANNEL_STATUSES";
    GatewayOpCodes[GatewayOpCodes["GUILD_SUBSCRIPTIONS_BULK"] = 37] = "GUILD_SUBSCRIPTIONS_BULK";
    GatewayOpCodes[GatewayOpCodes["GUILD_CHANNELS_RESYNC"] = 38] = "GUILD_CHANNELS_RESYNC";
    GatewayOpCodes[GatewayOpCodes["REQUEST_CHANNEL_MEMBER_COUNT"] = 39] = "REQUEST_CHANNEL_MEMBER_COUNT";
})(GatewayOpCodes || (GatewayOpCodes = {}));
