// asyncc/lib/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AsynccError", {
  enumerable: true,
  get: function get() {
    return _AsynccError["default"];
  }
});
Object.defineProperty(exports, "NoPromise", {
  enumerable: true,
  get: function get() {
    return _NoPromise["default"];
  }
});
Object.defineProperty(exports, "Queue", {
  enumerable: true,
  get: function get() {
    return _Queue["default"];
  }
});
Object.defineProperty(exports, "_setImmediate", {
  enumerable: true,
  get: function get() {
    return _setImmediate2._setImmediate;
  }
});
Object.defineProperty(exports, "compose", {
  enumerable: true,
  get: function get() {
    return _compose["default"];
  }
});
Object.defineProperty(exports, "connect", {
  enumerable: true,
  get: function get() {
    return _connect["default"];
  }
});
exports["default"] = void 0;
Object.defineProperty(exports, "doUntil", {
  enumerable: true,
  get: function get() {
    return _doUntil["default"];
  }
});
Object.defineProperty(exports, "doWhilst", {
  enumerable: true,
  get: function get() {
    return _doWhilst["default"];
  }
});
Object.defineProperty(exports, "each", {
  enumerable: true,
  get: function get() {
    return _each["default"];
  }
});
Object.defineProperty(exports, "eachLimit", {
  enumerable: true,
  get: function get() {
    return _eachLimit["default"];
  }
});
Object.defineProperty(exports, "eachSeries", {
  enumerable: true,
  get: function get() {
    return _eachSeries["default"];
  }
});
Object.defineProperty(exports, "noPromise", {
  enumerable: true,
  get: function get() {
    return _NoPromise.noPromise;
  }
});
Object.defineProperty(exports, "parallel", {
  enumerable: true,
  get: function get() {
    return _parallel["default"];
  }
});
Object.defineProperty(exports, "parallelLimit", {
  enumerable: true,
  get: function get() {
    return _parallelLimit["default"];
  }
});
Object.defineProperty(exports, "queue", {
  enumerable: true,
  get: function get() {
    return _Queue.queue;
  }
});
Object.defineProperty(exports, "retry", {
  enumerable: true,
  get: function get() {
    return _retry["default"];
  }
});
Object.defineProperty(exports, "series", {
  enumerable: true,
  get: function get() {
    return _series["default"];
  }
});
Object.defineProperty(exports, "times", {
  enumerable: true,
  get: function get() {
    return _times["default"];
  }
});
Object.defineProperty(exports, "until", {
  enumerable: true,
  get: function get() {
    return _until["default"];
  }
});
Object.defineProperty(exports, "whilst", {
  enumerable: true,
  get: function get() {
    return _whilst["default"];
  }
});
var _setImmediate2 = require("./_setImmediate");
var _AsynccError = _interopRequireDefault(require("./AsynccError"));
var _compose = _interopRequireDefault(require("./compose"));
var _connect = _interopRequireDefault(require("./connect"));
var _doUntil = _interopRequireDefault(require("./doUntil"));
var _doWhilst = _interopRequireDefault(require("./doWhilst"));
var _each = _interopRequireDefault(require("./each"));
var _eachLimit = _interopRequireDefault(require("./eachLimit"));
var _eachSeries = _interopRequireDefault(require("./eachSeries"));
var _NoPromise = _interopRequireWildcard(require("./NoPromise"));
var _parallel = _interopRequireDefault(require("./parallel"));
var _parallelLimit = _interopRequireDefault(require("./parallelLimit"));
var _Queue = _interopRequireWildcard(require("./Queue"));
var _retry = _interopRequireDefault(require("./retry"));
var _series = _interopRequireDefault(require("./series"));
var _times = _interopRequireDefault(require("./times"));
var _until = _interopRequireDefault(require("./until"));
var _whilst = _interopRequireDefault(require("./whilst"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
/**
* Serial execution patterns
* @module serial
*/
/**
* Parallel execution patterns
* @module parallel
*/
var _default = exports["default"] = {
  _setImmediate: _setImmediate2._setImmediate,
  AsynccError: _AsynccError["default"],
  compose: _compose["default"],
  connect: _connect["default"],
  doUntil: _doUntil["default"],
  doWhilst: _doWhilst["default"],
  each: _each["default"],
  eachLimit: _eachLimit["default"],
  eachSeries: _eachSeries["default"],
  noPromise: _NoPromise.noPromise,
  NoPromise: _NoPromise["default"],
  parallel: _parallel["default"],
  parallelLimit: _parallelLimit["default"],
  queue: _Queue.queue,
  Queue: _Queue["default"],
  retry: _retry["default"],
  series: _series["default"],
  times: _times["default"],
  until: _until["default"],
  whilst: _whilst["default"]
};
