// @dank074/discord-video-stream/dist/media/LibavCodecId.js
// https://ffmpeg.org/doxygen/7.0/codec__id_8h_source.html
// biome-ignore lint/style/useEnumInitializers: copied from upstream ffmpeg
export var AVCodecID;
(function (AVCodecID) {
    AVCodecID[AVCodecID["AV_CODEC_ID_NONE"] = 0] = "AV_CODEC_ID_NONE";
    /* video codecs */
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEG1VIDEO"] = 1] = "AV_CODEC_ID_MPEG1VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEG2VIDEO"] = 2] = "AV_CODEC_ID_MPEG2VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_H261"] = 3] = "AV_CODEC_ID_H261";
    AVCodecID[AVCodecID["AV_CODEC_ID_H263"] = 4] = "AV_CODEC_ID_H263";
    AVCodecID[AVCodecID["AV_CODEC_ID_RV10"] = 5] = "AV_CODEC_ID_RV10";
    AVCodecID[AVCodecID["AV_CODEC_ID_RV20"] = 6] = "AV_CODEC_ID_RV20";
    AVCodecID[AVCodecID["AV_CODEC_ID_MJPEG"] = 7] = "AV_CODEC_ID_MJPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_MJPEGB"] = 8] = "AV_CODEC_ID_MJPEGB";
    AVCodecID[AVCodecID["AV_CODEC_ID_LJPEG"] = 9] = "AV_CODEC_ID_LJPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_SP5X"] = 10] = "AV_CODEC_ID_SP5X";
    AVCodecID[AVCodecID["AV_CODEC_ID_JPEGLS"] = 11] = "AV_CODEC_ID_JPEGLS";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEG4"] = 12] = "AV_CODEC_ID_MPEG4";
    AVCodecID[AVCodecID["AV_CODEC_ID_RAWVIDEO"] = 13] = "AV_CODEC_ID_RAWVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSMPEG4V1"] = 14] = "AV_CODEC_ID_MSMPEG4V1";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSMPEG4V2"] = 15] = "AV_CODEC_ID_MSMPEG4V2";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSMPEG4V3"] = 16] = "AV_CODEC_ID_MSMPEG4V3";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMV1"] = 17] = "AV_CODEC_ID_WMV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMV2"] = 18] = "AV_CODEC_ID_WMV2";
    AVCodecID[AVCodecID["AV_CODEC_ID_H263P"] = 19] = "AV_CODEC_ID_H263P";
    AVCodecID[AVCodecID["AV_CODEC_ID_H263I"] = 20] = "AV_CODEC_ID_H263I";
    AVCodecID[AVCodecID["AV_CODEC_ID_FLV1"] = 21] = "AV_CODEC_ID_FLV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_SVQ1"] = 22] = "AV_CODEC_ID_SVQ1";
    AVCodecID[AVCodecID["AV_CODEC_ID_SVQ3"] = 23] = "AV_CODEC_ID_SVQ3";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVVIDEO"] = 24] = "AV_CODEC_ID_DVVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_HUFFYUV"] = 25] = "AV_CODEC_ID_HUFFYUV";
    AVCodecID[AVCodecID["AV_CODEC_ID_CYUV"] = 26] = "AV_CODEC_ID_CYUV";
    AVCodecID[AVCodecID["AV_CODEC_ID_H264"] = 27] = "AV_CODEC_ID_H264";
    AVCodecID[AVCodecID["AV_CODEC_ID_INDEO3"] = 28] = "AV_CODEC_ID_INDEO3";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP3"] = 29] = "AV_CODEC_ID_VP3";
    AVCodecID[AVCodecID["AV_CODEC_ID_THEORA"] = 30] = "AV_CODEC_ID_THEORA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ASV1"] = 31] = "AV_CODEC_ID_ASV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_ASV2"] = 32] = "AV_CODEC_ID_ASV2";
    AVCodecID[AVCodecID["AV_CODEC_ID_FFV1"] = 33] = "AV_CODEC_ID_FFV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_4XM"] = 34] = "AV_CODEC_ID_4XM";
    AVCodecID[AVCodecID["AV_CODEC_ID_VCR1"] = 35] = "AV_CODEC_ID_VCR1";
    AVCodecID[AVCodecID["AV_CODEC_ID_CLJR"] = 36] = "AV_CODEC_ID_CLJR";
    AVCodecID[AVCodecID["AV_CODEC_ID_MDEC"] = 37] = "AV_CODEC_ID_MDEC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ROQ"] = 38] = "AV_CODEC_ID_ROQ";
    AVCodecID[AVCodecID["AV_CODEC_ID_INTERPLAY_VIDEO"] = 39] = "AV_CODEC_ID_INTERPLAY_VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_XAN_WC3"] = 40] = "AV_CODEC_ID_XAN_WC3";
    AVCodecID[AVCodecID["AV_CODEC_ID_XAN_WC4"] = 41] = "AV_CODEC_ID_XAN_WC4";
    AVCodecID[AVCodecID["AV_CODEC_ID_RPZA"] = 42] = "AV_CODEC_ID_RPZA";
    AVCodecID[AVCodecID["AV_CODEC_ID_CINEPAK"] = 43] = "AV_CODEC_ID_CINEPAK";
    AVCodecID[AVCodecID["AV_CODEC_ID_WS_VQA"] = 44] = "AV_CODEC_ID_WS_VQA";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSRLE"] = 45] = "AV_CODEC_ID_MSRLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSVIDEO1"] = 46] = "AV_CODEC_ID_MSVIDEO1";
    AVCodecID[AVCodecID["AV_CODEC_ID_IDCIN"] = 47] = "AV_CODEC_ID_IDCIN";
    AVCodecID[AVCodecID["AV_CODEC_ID_8BPS"] = 48] = "AV_CODEC_ID_8BPS";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMC"] = 49] = "AV_CODEC_ID_SMC";
    AVCodecID[AVCodecID["AV_CODEC_ID_FLIC"] = 50] = "AV_CODEC_ID_FLIC";
    AVCodecID[AVCodecID["AV_CODEC_ID_TRUEMOTION1"] = 51] = "AV_CODEC_ID_TRUEMOTION1";
    AVCodecID[AVCodecID["AV_CODEC_ID_VMDVIDEO"] = 52] = "AV_CODEC_ID_VMDVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSZH"] = 53] = "AV_CODEC_ID_MSZH";
    AVCodecID[AVCodecID["AV_CODEC_ID_ZLIB"] = 54] = "AV_CODEC_ID_ZLIB";
    AVCodecID[AVCodecID["AV_CODEC_ID_QTRLE"] = 55] = "AV_CODEC_ID_QTRLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_TSCC"] = 56] = "AV_CODEC_ID_TSCC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ULTI"] = 57] = "AV_CODEC_ID_ULTI";
    AVCodecID[AVCodecID["AV_CODEC_ID_QDRAW"] = 58] = "AV_CODEC_ID_QDRAW";
    AVCodecID[AVCodecID["AV_CODEC_ID_VIXL"] = 59] = "AV_CODEC_ID_VIXL";
    AVCodecID[AVCodecID["AV_CODEC_ID_QPEG"] = 60] = "AV_CODEC_ID_QPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_PNG"] = 61] = "AV_CODEC_ID_PNG";
    AVCodecID[AVCodecID["AV_CODEC_ID_PPM"] = 62] = "AV_CODEC_ID_PPM";
    AVCodecID[AVCodecID["AV_CODEC_ID_PBM"] = 63] = "AV_CODEC_ID_PBM";
    AVCodecID[AVCodecID["AV_CODEC_ID_PGM"] = 64] = "AV_CODEC_ID_PGM";
    AVCodecID[AVCodecID["AV_CODEC_ID_PGMYUV"] = 65] = "AV_CODEC_ID_PGMYUV";
    AVCodecID[AVCodecID["AV_CODEC_ID_PAM"] = 66] = "AV_CODEC_ID_PAM";
    AVCodecID[AVCodecID["AV_CODEC_ID_FFVHUFF"] = 67] = "AV_CODEC_ID_FFVHUFF";
    AVCodecID[AVCodecID["AV_CODEC_ID_RV30"] = 68] = "AV_CODEC_ID_RV30";
    AVCodecID[AVCodecID["AV_CODEC_ID_RV40"] = 69] = "AV_CODEC_ID_RV40";
    AVCodecID[AVCodecID["AV_CODEC_ID_VC1"] = 70] = "AV_CODEC_ID_VC1";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMV3"] = 71] = "AV_CODEC_ID_WMV3";
    AVCodecID[AVCodecID["AV_CODEC_ID_LOCO"] = 72] = "AV_CODEC_ID_LOCO";
    AVCodecID[AVCodecID["AV_CODEC_ID_WNV1"] = 73] = "AV_CODEC_ID_WNV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_AASC"] = 74] = "AV_CODEC_ID_AASC";
    AVCodecID[AVCodecID["AV_CODEC_ID_INDEO2"] = 75] = "AV_CODEC_ID_INDEO2";
    AVCodecID[AVCodecID["AV_CODEC_ID_FRAPS"] = 76] = "AV_CODEC_ID_FRAPS";
    AVCodecID[AVCodecID["AV_CODEC_ID_TRUEMOTION2"] = 77] = "AV_CODEC_ID_TRUEMOTION2";
    AVCodecID[AVCodecID["AV_CODEC_ID_BMP"] = 78] = "AV_CODEC_ID_BMP";
    AVCodecID[AVCodecID["AV_CODEC_ID_CSCD"] = 79] = "AV_CODEC_ID_CSCD";
    AVCodecID[AVCodecID["AV_CODEC_ID_MMVIDEO"] = 80] = "AV_CODEC_ID_MMVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_ZMBV"] = 81] = "AV_CODEC_ID_ZMBV";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVS"] = 82] = "AV_CODEC_ID_AVS";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMACKVIDEO"] = 83] = "AV_CODEC_ID_SMACKVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_NUV"] = 84] = "AV_CODEC_ID_NUV";
    AVCodecID[AVCodecID["AV_CODEC_ID_KMVC"] = 85] = "AV_CODEC_ID_KMVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_FLASHSV"] = 86] = "AV_CODEC_ID_FLASHSV";
    AVCodecID[AVCodecID["AV_CODEC_ID_CAVS"] = 87] = "AV_CODEC_ID_CAVS";
    AVCodecID[AVCodecID["AV_CODEC_ID_JPEG2000"] = 88] = "AV_CODEC_ID_JPEG2000";
    AVCodecID[AVCodecID["AV_CODEC_ID_VMNC"] = 89] = "AV_CODEC_ID_VMNC";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP5"] = 90] = "AV_CODEC_ID_VP5";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP6"] = 91] = "AV_CODEC_ID_VP6";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP6F"] = 92] = "AV_CODEC_ID_VP6F";
    AVCodecID[AVCodecID["AV_CODEC_ID_TARGA"] = 93] = "AV_CODEC_ID_TARGA";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSICINVIDEO"] = 94] = "AV_CODEC_ID_DSICINVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_TIERTEXSEQVIDEO"] = 95] = "AV_CODEC_ID_TIERTEXSEQVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_TIFF"] = 96] = "AV_CODEC_ID_TIFF";
    AVCodecID[AVCodecID["AV_CODEC_ID_GIF"] = 97] = "AV_CODEC_ID_GIF";
    AVCodecID[AVCodecID["AV_CODEC_ID_DXA"] = 98] = "AV_CODEC_ID_DXA";
    AVCodecID[AVCodecID["AV_CODEC_ID_DNXHD"] = 99] = "AV_CODEC_ID_DNXHD";
    AVCodecID[AVCodecID["AV_CODEC_ID_THP"] = 100] = "AV_CODEC_ID_THP";
    AVCodecID[AVCodecID["AV_CODEC_ID_SGI"] = 101] = "AV_CODEC_ID_SGI";
    AVCodecID[AVCodecID["AV_CODEC_ID_C93"] = 102] = "AV_CODEC_ID_C93";
    AVCodecID[AVCodecID["AV_CODEC_ID_BETHSOFTVID"] = 103] = "AV_CODEC_ID_BETHSOFTVID";
    AVCodecID[AVCodecID["AV_CODEC_ID_PTX"] = 104] = "AV_CODEC_ID_PTX";
    AVCodecID[AVCodecID["AV_CODEC_ID_TXD"] = 105] = "AV_CODEC_ID_TXD";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP6A"] = 106] = "AV_CODEC_ID_VP6A";
    AVCodecID[AVCodecID["AV_CODEC_ID_AMV"] = 107] = "AV_CODEC_ID_AMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_VB"] = 108] = "AV_CODEC_ID_VB";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCX"] = 109] = "AV_CODEC_ID_PCX";
    AVCodecID[AVCodecID["AV_CODEC_ID_SUNRAST"] = 110] = "AV_CODEC_ID_SUNRAST";
    AVCodecID[AVCodecID["AV_CODEC_ID_INDEO4"] = 111] = "AV_CODEC_ID_INDEO4";
    AVCodecID[AVCodecID["AV_CODEC_ID_INDEO5"] = 112] = "AV_CODEC_ID_INDEO5";
    AVCodecID[AVCodecID["AV_CODEC_ID_MIMIC"] = 113] = "AV_CODEC_ID_MIMIC";
    AVCodecID[AVCodecID["AV_CODEC_ID_RL2"] = 114] = "AV_CODEC_ID_RL2";
    AVCodecID[AVCodecID["AV_CODEC_ID_ESCAPE124"] = 115] = "AV_CODEC_ID_ESCAPE124";
    AVCodecID[AVCodecID["AV_CODEC_ID_DIRAC"] = 116] = "AV_CODEC_ID_DIRAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_BFI"] = 117] = "AV_CODEC_ID_BFI";
    AVCodecID[AVCodecID["AV_CODEC_ID_CMV"] = 118] = "AV_CODEC_ID_CMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_MOTIONPIXELS"] = 119] = "AV_CODEC_ID_MOTIONPIXELS";
    AVCodecID[AVCodecID["AV_CODEC_ID_TGV"] = 120] = "AV_CODEC_ID_TGV";
    AVCodecID[AVCodecID["AV_CODEC_ID_TGQ"] = 121] = "AV_CODEC_ID_TGQ";
    AVCodecID[AVCodecID["AV_CODEC_ID_TQI"] = 122] = "AV_CODEC_ID_TQI";
    AVCodecID[AVCodecID["AV_CODEC_ID_AURA"] = 123] = "AV_CODEC_ID_AURA";
    AVCodecID[AVCodecID["AV_CODEC_ID_AURA2"] = 124] = "AV_CODEC_ID_AURA2";
    AVCodecID[AVCodecID["AV_CODEC_ID_V210X"] = 125] = "AV_CODEC_ID_V210X";
    AVCodecID[AVCodecID["AV_CODEC_ID_TMV"] = 126] = "AV_CODEC_ID_TMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_V210"] = 127] = "AV_CODEC_ID_V210";
    AVCodecID[AVCodecID["AV_CODEC_ID_DPX"] = 128] = "AV_CODEC_ID_DPX";
    AVCodecID[AVCodecID["AV_CODEC_ID_MAD"] = 129] = "AV_CODEC_ID_MAD";
    AVCodecID[AVCodecID["AV_CODEC_ID_FRWU"] = 130] = "AV_CODEC_ID_FRWU";
    AVCodecID[AVCodecID["AV_CODEC_ID_FLASHSV2"] = 131] = "AV_CODEC_ID_FLASHSV2";
    AVCodecID[AVCodecID["AV_CODEC_ID_CDGRAPHICS"] = 132] = "AV_CODEC_ID_CDGRAPHICS";
    AVCodecID[AVCodecID["AV_CODEC_ID_R210"] = 133] = "AV_CODEC_ID_R210";
    AVCodecID[AVCodecID["AV_CODEC_ID_ANM"] = 134] = "AV_CODEC_ID_ANM";
    AVCodecID[AVCodecID["AV_CODEC_ID_BINKVIDEO"] = 135] = "AV_CODEC_ID_BINKVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_IFF_ILBM"] = 136] = "AV_CODEC_ID_IFF_ILBM";
    AVCodecID[AVCodecID["AV_CODEC_ID_IFF_BYTERUN1"] = 136] = "AV_CODEC_ID_IFF_BYTERUN1";
    AVCodecID[AVCodecID["AV_CODEC_ID_KGV1"] = 137] = "AV_CODEC_ID_KGV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_YOP"] = 138] = "AV_CODEC_ID_YOP";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP8"] = 139] = "AV_CODEC_ID_VP8";
    AVCodecID[AVCodecID["AV_CODEC_ID_PICTOR"] = 140] = "AV_CODEC_ID_PICTOR";
    AVCodecID[AVCodecID["AV_CODEC_ID_ANSI"] = 141] = "AV_CODEC_ID_ANSI";
    AVCodecID[AVCodecID["AV_CODEC_ID_A64_MULTI"] = 142] = "AV_CODEC_ID_A64_MULTI";
    AVCodecID[AVCodecID["AV_CODEC_ID_A64_MULTI5"] = 143] = "AV_CODEC_ID_A64_MULTI5";
    AVCodecID[AVCodecID["AV_CODEC_ID_R10K"] = 144] = "AV_CODEC_ID_R10K";
    AVCodecID[AVCodecID["AV_CODEC_ID_MXPEG"] = 145] = "AV_CODEC_ID_MXPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_LAGARITH"] = 146] = "AV_CODEC_ID_LAGARITH";
    AVCodecID[AVCodecID["AV_CODEC_ID_PRORES"] = 147] = "AV_CODEC_ID_PRORES";
    AVCodecID[AVCodecID["AV_CODEC_ID_JV"] = 148] = "AV_CODEC_ID_JV";
    AVCodecID[AVCodecID["AV_CODEC_ID_DFA"] = 149] = "AV_CODEC_ID_DFA";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMV3IMAGE"] = 150] = "AV_CODEC_ID_WMV3IMAGE";
    AVCodecID[AVCodecID["AV_CODEC_ID_VC1IMAGE"] = 151] = "AV_CODEC_ID_VC1IMAGE";
    AVCodecID[AVCodecID["AV_CODEC_ID_UTVIDEO"] = 152] = "AV_CODEC_ID_UTVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_BMV_VIDEO"] = 153] = "AV_CODEC_ID_BMV_VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_VBLE"] = 154] = "AV_CODEC_ID_VBLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_DXTORY"] = 155] = "AV_CODEC_ID_DXTORY";
    AVCodecID[AVCodecID["AV_CODEC_ID_V410"] = 156] = "AV_CODEC_ID_V410";
    AVCodecID[AVCodecID["AV_CODEC_ID_XWD"] = 157] = "AV_CODEC_ID_XWD";
    AVCodecID[AVCodecID["AV_CODEC_ID_CDXL"] = 158] = "AV_CODEC_ID_CDXL";
    AVCodecID[AVCodecID["AV_CODEC_ID_XBM"] = 159] = "AV_CODEC_ID_XBM";
    AVCodecID[AVCodecID["AV_CODEC_ID_ZEROCODEC"] = 160] = "AV_CODEC_ID_ZEROCODEC";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSS1"] = 161] = "AV_CODEC_ID_MSS1";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSA1"] = 162] = "AV_CODEC_ID_MSA1";
    AVCodecID[AVCodecID["AV_CODEC_ID_TSCC2"] = 163] = "AV_CODEC_ID_TSCC2";
    AVCodecID[AVCodecID["AV_CODEC_ID_MTS2"] = 164] = "AV_CODEC_ID_MTS2";
    AVCodecID[AVCodecID["AV_CODEC_ID_CLLC"] = 165] = "AV_CODEC_ID_CLLC";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSS2"] = 166] = "AV_CODEC_ID_MSS2";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP9"] = 167] = "AV_CODEC_ID_VP9";
    AVCodecID[AVCodecID["AV_CODEC_ID_AIC"] = 168] = "AV_CODEC_ID_AIC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ESCAPE130"] = 169] = "AV_CODEC_ID_ESCAPE130";
    AVCodecID[AVCodecID["AV_CODEC_ID_G2M"] = 170] = "AV_CODEC_ID_G2M";
    AVCodecID[AVCodecID["AV_CODEC_ID_WEBP"] = 171] = "AV_CODEC_ID_WEBP";
    AVCodecID[AVCodecID["AV_CODEC_ID_HNM4_VIDEO"] = 172] = "AV_CODEC_ID_HNM4_VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_HEVC"] = 173] = "AV_CODEC_ID_HEVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_H265"] = 173] = "AV_CODEC_ID_H265";
    AVCodecID[AVCodecID["AV_CODEC_ID_FIC"] = 174] = "AV_CODEC_ID_FIC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ALIAS_PIX"] = 175] = "AV_CODEC_ID_ALIAS_PIX";
    AVCodecID[AVCodecID["AV_CODEC_ID_BRENDER_PIX"] = 176] = "AV_CODEC_ID_BRENDER_PIX";
    AVCodecID[AVCodecID["AV_CODEC_ID_PAF_VIDEO"] = 177] = "AV_CODEC_ID_PAF_VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_EXR"] = 178] = "AV_CODEC_ID_EXR";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP7"] = 179] = "AV_CODEC_ID_VP7";
    AVCodecID[AVCodecID["AV_CODEC_ID_SANM"] = 180] = "AV_CODEC_ID_SANM";
    AVCodecID[AVCodecID["AV_CODEC_ID_SGIRLE"] = 181] = "AV_CODEC_ID_SGIRLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_MVC1"] = 182] = "AV_CODEC_ID_MVC1";
    AVCodecID[AVCodecID["AV_CODEC_ID_MVC2"] = 183] = "AV_CODEC_ID_MVC2";
    AVCodecID[AVCodecID["AV_CODEC_ID_HQX"] = 184] = "AV_CODEC_ID_HQX";
    AVCodecID[AVCodecID["AV_CODEC_ID_TDSC"] = 185] = "AV_CODEC_ID_TDSC";
    AVCodecID[AVCodecID["AV_CODEC_ID_HQ_HQA"] = 186] = "AV_CODEC_ID_HQ_HQA";
    AVCodecID[AVCodecID["AV_CODEC_ID_HAP"] = 187] = "AV_CODEC_ID_HAP";
    AVCodecID[AVCodecID["AV_CODEC_ID_DDS"] = 188] = "AV_CODEC_ID_DDS";
    AVCodecID[AVCodecID["AV_CODEC_ID_DXV"] = 189] = "AV_CODEC_ID_DXV";
    AVCodecID[AVCodecID["AV_CODEC_ID_SCREENPRESSO"] = 190] = "AV_CODEC_ID_SCREENPRESSO";
    AVCodecID[AVCodecID["AV_CODEC_ID_RSCC"] = 191] = "AV_CODEC_ID_RSCC";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVS2"] = 192] = "AV_CODEC_ID_AVS2";
    AVCodecID[AVCodecID["AV_CODEC_ID_PGX"] = 193] = "AV_CODEC_ID_PGX";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVS3"] = 194] = "AV_CODEC_ID_AVS3";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSP2"] = 195] = "AV_CODEC_ID_MSP2";
    AVCodecID[AVCodecID["AV_CODEC_ID_VVC"] = 196] = "AV_CODEC_ID_VVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_H266"] = 196] = "AV_CODEC_ID_H266";
    AVCodecID[AVCodecID["AV_CODEC_ID_Y41P"] = 197] = "AV_CODEC_ID_Y41P";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVRP"] = 198] = "AV_CODEC_ID_AVRP";
    AVCodecID[AVCodecID["AV_CODEC_ID_012V"] = 199] = "AV_CODEC_ID_012V";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVUI"] = 200] = "AV_CODEC_ID_AVUI";
    AVCodecID[AVCodecID["AV_CODEC_ID_TARGA_Y216"] = 201] = "AV_CODEC_ID_TARGA_Y216";
    AVCodecID[AVCodecID["AV_CODEC_ID_V308"] = 202] = "AV_CODEC_ID_V308";
    AVCodecID[AVCodecID["AV_CODEC_ID_V408"] = 203] = "AV_CODEC_ID_V408";
    AVCodecID[AVCodecID["AV_CODEC_ID_YUV4"] = 204] = "AV_CODEC_ID_YUV4";
    AVCodecID[AVCodecID["AV_CODEC_ID_AVRN"] = 205] = "AV_CODEC_ID_AVRN";
    AVCodecID[AVCodecID["AV_CODEC_ID_CPIA"] = 206] = "AV_CODEC_ID_CPIA";
    AVCodecID[AVCodecID["AV_CODEC_ID_XFACE"] = 207] = "AV_CODEC_ID_XFACE";
    AVCodecID[AVCodecID["AV_CODEC_ID_SNOW"] = 208] = "AV_CODEC_ID_SNOW";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMVJPEG"] = 209] = "AV_CODEC_ID_SMVJPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_APNG"] = 210] = "AV_CODEC_ID_APNG";
    AVCodecID[AVCodecID["AV_CODEC_ID_DAALA"] = 211] = "AV_CODEC_ID_DAALA";
    AVCodecID[AVCodecID["AV_CODEC_ID_CFHD"] = 212] = "AV_CODEC_ID_CFHD";
    AVCodecID[AVCodecID["AV_CODEC_ID_TRUEMOTION2RT"] = 213] = "AV_CODEC_ID_TRUEMOTION2RT";
    AVCodecID[AVCodecID["AV_CODEC_ID_M101"] = 214] = "AV_CODEC_ID_M101";
    AVCodecID[AVCodecID["AV_CODEC_ID_MAGICYUV"] = 215] = "AV_CODEC_ID_MAGICYUV";
    AVCodecID[AVCodecID["AV_CODEC_ID_SHEERVIDEO"] = 216] = "AV_CODEC_ID_SHEERVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_YLC"] = 217] = "AV_CODEC_ID_YLC";
    AVCodecID[AVCodecID["AV_CODEC_ID_PSD"] = 218] = "AV_CODEC_ID_PSD";
    AVCodecID[AVCodecID["AV_CODEC_ID_PIXLET"] = 219] = "AV_CODEC_ID_PIXLET";
    AVCodecID[AVCodecID["AV_CODEC_ID_SPEEDHQ"] = 220] = "AV_CODEC_ID_SPEEDHQ";
    AVCodecID[AVCodecID["AV_CODEC_ID_FMVC"] = 221] = "AV_CODEC_ID_FMVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_SCPR"] = 222] = "AV_CODEC_ID_SCPR";
    AVCodecID[AVCodecID["AV_CODEC_ID_CLEARVIDEO"] = 223] = "AV_CODEC_ID_CLEARVIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_XPM"] = 224] = "AV_CODEC_ID_XPM";
    AVCodecID[AVCodecID["AV_CODEC_ID_AV1"] = 225] = "AV_CODEC_ID_AV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_BITPACKED"] = 226] = "AV_CODEC_ID_BITPACKED";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSCC"] = 227] = "AV_CODEC_ID_MSCC";
    AVCodecID[AVCodecID["AV_CODEC_ID_SRGC"] = 228] = "AV_CODEC_ID_SRGC";
    AVCodecID[AVCodecID["AV_CODEC_ID_SVG"] = 229] = "AV_CODEC_ID_SVG";
    AVCodecID[AVCodecID["AV_CODEC_ID_GDV"] = 230] = "AV_CODEC_ID_GDV";
    AVCodecID[AVCodecID["AV_CODEC_ID_FITS"] = 231] = "AV_CODEC_ID_FITS";
    AVCodecID[AVCodecID["AV_CODEC_ID_IMM4"] = 232] = "AV_CODEC_ID_IMM4";
    AVCodecID[AVCodecID["AV_CODEC_ID_PROSUMER"] = 233] = "AV_CODEC_ID_PROSUMER";
    AVCodecID[AVCodecID["AV_CODEC_ID_MWSC"] = 234] = "AV_CODEC_ID_MWSC";
    AVCodecID[AVCodecID["AV_CODEC_ID_WCMV"] = 235] = "AV_CODEC_ID_WCMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_RASC"] = 236] = "AV_CODEC_ID_RASC";
    AVCodecID[AVCodecID["AV_CODEC_ID_HYMT"] = 237] = "AV_CODEC_ID_HYMT";
    AVCodecID[AVCodecID["AV_CODEC_ID_ARBC"] = 238] = "AV_CODEC_ID_ARBC";
    AVCodecID[AVCodecID["AV_CODEC_ID_AGM"] = 239] = "AV_CODEC_ID_AGM";
    AVCodecID[AVCodecID["AV_CODEC_ID_LSCR"] = 240] = "AV_CODEC_ID_LSCR";
    AVCodecID[AVCodecID["AV_CODEC_ID_VP4"] = 241] = "AV_CODEC_ID_VP4";
    AVCodecID[AVCodecID["AV_CODEC_ID_IMM5"] = 242] = "AV_CODEC_ID_IMM5";
    AVCodecID[AVCodecID["AV_CODEC_ID_MVDV"] = 243] = "AV_CODEC_ID_MVDV";
    AVCodecID[AVCodecID["AV_CODEC_ID_MVHA"] = 244] = "AV_CODEC_ID_MVHA";
    AVCodecID[AVCodecID["AV_CODEC_ID_CDTOONS"] = 245] = "AV_CODEC_ID_CDTOONS";
    AVCodecID[AVCodecID["AV_CODEC_ID_MV30"] = 246] = "AV_CODEC_ID_MV30";
    AVCodecID[AVCodecID["AV_CODEC_ID_NOTCHLC"] = 247] = "AV_CODEC_ID_NOTCHLC";
    AVCodecID[AVCodecID["AV_CODEC_ID_PFM"] = 248] = "AV_CODEC_ID_PFM";
    AVCodecID[AVCodecID["AV_CODEC_ID_MOBICLIP"] = 249] = "AV_CODEC_ID_MOBICLIP";
    AVCodecID[AVCodecID["AV_CODEC_ID_PHOTOCD"] = 250] = "AV_CODEC_ID_PHOTOCD";
    AVCodecID[AVCodecID["AV_CODEC_ID_IPU"] = 251] = "AV_CODEC_ID_IPU";
    AVCodecID[AVCodecID["AV_CODEC_ID_ARGO"] = 252] = "AV_CODEC_ID_ARGO";
    AVCodecID[AVCodecID["AV_CODEC_ID_CRI"] = 253] = "AV_CODEC_ID_CRI";
    AVCodecID[AVCodecID["AV_CODEC_ID_SIMBIOSIS_IMX"] = 254] = "AV_CODEC_ID_SIMBIOSIS_IMX";
    AVCodecID[AVCodecID["AV_CODEC_ID_SGA_VIDEO"] = 255] = "AV_CODEC_ID_SGA_VIDEO";
    AVCodecID[AVCodecID["AV_CODEC_ID_GEM"] = 256] = "AV_CODEC_ID_GEM";
    AVCodecID[AVCodecID["AV_CODEC_ID_VBN"] = 257] = "AV_CODEC_ID_VBN";
    AVCodecID[AVCodecID["AV_CODEC_ID_JPEGXL"] = 258] = "AV_CODEC_ID_JPEGXL";
    AVCodecID[AVCodecID["AV_CODEC_ID_QOI"] = 259] = "AV_CODEC_ID_QOI";
    AVCodecID[AVCodecID["AV_CODEC_ID_PHM"] = 260] = "AV_CODEC_ID_PHM";
    AVCodecID[AVCodecID["AV_CODEC_ID_RADIANCE_HDR"] = 261] = "AV_CODEC_ID_RADIANCE_HDR";
    AVCodecID[AVCodecID["AV_CODEC_ID_WBMP"] = 262] = "AV_CODEC_ID_WBMP";
    AVCodecID[AVCodecID["AV_CODEC_ID_MEDIA100"] = 263] = "AV_CODEC_ID_MEDIA100";
    AVCodecID[AVCodecID["AV_CODEC_ID_VQC"] = 264] = "AV_CODEC_ID_VQC";
    AVCodecID[AVCodecID["AV_CODEC_ID_PDV"] = 265] = "AV_CODEC_ID_PDV";
    AVCodecID[AVCodecID["AV_CODEC_ID_EVC"] = 266] = "AV_CODEC_ID_EVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_RTV1"] = 267] = "AV_CODEC_ID_RTV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_VMIX"] = 268] = "AV_CODEC_ID_VMIX";
    AVCodecID[AVCodecID["AV_CODEC_ID_LEAD"] = 269] = "AV_CODEC_ID_LEAD";
    /* various PCM "codecs" */
    AVCodecID[AVCodecID["AV_CODEC_ID_FIRST_AUDIO"] = 65536] = "AV_CODEC_ID_FIRST_AUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S16LE"] = 65536] = "AV_CODEC_ID_PCM_S16LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S16BE"] = 65537] = "AV_CODEC_ID_PCM_S16BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U16LE"] = 65538] = "AV_CODEC_ID_PCM_U16LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U16BE"] = 65539] = "AV_CODEC_ID_PCM_U16BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S8"] = 65540] = "AV_CODEC_ID_PCM_S8";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U8"] = 65541] = "AV_CODEC_ID_PCM_U8";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_MULAW"] = 65542] = "AV_CODEC_ID_PCM_MULAW";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_ALAW"] = 65543] = "AV_CODEC_ID_PCM_ALAW";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S32LE"] = 65544] = "AV_CODEC_ID_PCM_S32LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S32BE"] = 65545] = "AV_CODEC_ID_PCM_S32BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U32LE"] = 65546] = "AV_CODEC_ID_PCM_U32LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U32BE"] = 65547] = "AV_CODEC_ID_PCM_U32BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S24LE"] = 65548] = "AV_CODEC_ID_PCM_S24LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S24BE"] = 65549] = "AV_CODEC_ID_PCM_S24BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U24LE"] = 65550] = "AV_CODEC_ID_PCM_U24LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_U24BE"] = 65551] = "AV_CODEC_ID_PCM_U24BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S24DAUD"] = 65552] = "AV_CODEC_ID_PCM_S24DAUD";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_ZORK"] = 65553] = "AV_CODEC_ID_PCM_ZORK";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S16LE_PLANAR"] = 65554] = "AV_CODEC_ID_PCM_S16LE_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_DVD"] = 65555] = "AV_CODEC_ID_PCM_DVD";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F32BE"] = 65556] = "AV_CODEC_ID_PCM_F32BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F32LE"] = 65557] = "AV_CODEC_ID_PCM_F32LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F64BE"] = 65558] = "AV_CODEC_ID_PCM_F64BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F64LE"] = 65559] = "AV_CODEC_ID_PCM_F64LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_BLURAY"] = 65560] = "AV_CODEC_ID_PCM_BLURAY";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_LXF"] = 65561] = "AV_CODEC_ID_PCM_LXF";
    AVCodecID[AVCodecID["AV_CODEC_ID_S302M"] = 65562] = "AV_CODEC_ID_S302M";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S8_PLANAR"] = 65563] = "AV_CODEC_ID_PCM_S8_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S24LE_PLANAR"] = 65564] = "AV_CODEC_ID_PCM_S24LE_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S32LE_PLANAR"] = 65565] = "AV_CODEC_ID_PCM_S32LE_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S16BE_PLANAR"] = 65566] = "AV_CODEC_ID_PCM_S16BE_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S64LE"] = 65567] = "AV_CODEC_ID_PCM_S64LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_S64BE"] = 65568] = "AV_CODEC_ID_PCM_S64BE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F16LE"] = 65569] = "AV_CODEC_ID_PCM_F16LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_F24LE"] = 65570] = "AV_CODEC_ID_PCM_F24LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_VIDC"] = 65571] = "AV_CODEC_ID_PCM_VIDC";
    AVCodecID[AVCodecID["AV_CODEC_ID_PCM_SGA"] = 65572] = "AV_CODEC_ID_PCM_SGA";
    /* various ADPCM codecs */
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_QT"] = 69632] = "AV_CODEC_ID_ADPCM_IMA_QT";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_WAV"] = 69633] = "AV_CODEC_ID_ADPCM_IMA_WAV";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_DK3"] = 69634] = "AV_CODEC_ID_ADPCM_IMA_DK3";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_DK4"] = 69635] = "AV_CODEC_ID_ADPCM_IMA_DK4";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_WS"] = 69636] = "AV_CODEC_ID_ADPCM_IMA_WS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_SMJPEG"] = 69637] = "AV_CODEC_ID_ADPCM_IMA_SMJPEG";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_MS"] = 69638] = "AV_CODEC_ID_ADPCM_MS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_4XM"] = 69639] = "AV_CODEC_ID_ADPCM_4XM";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_XA"] = 69640] = "AV_CODEC_ID_ADPCM_XA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_ADX"] = 69641] = "AV_CODEC_ID_ADPCM_ADX";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA"] = 69642] = "AV_CODEC_ID_ADPCM_EA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_G726"] = 69643] = "AV_CODEC_ID_ADPCM_G726";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_CT"] = 69644] = "AV_CODEC_ID_ADPCM_CT";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_SWF"] = 69645] = "AV_CODEC_ID_ADPCM_SWF";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_YAMAHA"] = 69646] = "AV_CODEC_ID_ADPCM_YAMAHA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_SBPRO_4"] = 69647] = "AV_CODEC_ID_ADPCM_SBPRO_4";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_SBPRO_3"] = 69648] = "AV_CODEC_ID_ADPCM_SBPRO_3";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_SBPRO_2"] = 69649] = "AV_CODEC_ID_ADPCM_SBPRO_2";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_THP"] = 69650] = "AV_CODEC_ID_ADPCM_THP";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_AMV"] = 69651] = "AV_CODEC_ID_ADPCM_IMA_AMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA_R1"] = 69652] = "AV_CODEC_ID_ADPCM_EA_R1";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA_R3"] = 69653] = "AV_CODEC_ID_ADPCM_EA_R3";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA_R2"] = 69654] = "AV_CODEC_ID_ADPCM_EA_R2";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_EA_SEAD"] = 69655] = "AV_CODEC_ID_ADPCM_IMA_EA_SEAD";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_EA_EACS"] = 69656] = "AV_CODEC_ID_ADPCM_IMA_EA_EACS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA_XAS"] = 69657] = "AV_CODEC_ID_ADPCM_EA_XAS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_EA_MAXIS_XA"] = 69658] = "AV_CODEC_ID_ADPCM_EA_MAXIS_XA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_ISS"] = 69659] = "AV_CODEC_ID_ADPCM_IMA_ISS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_G722"] = 69660] = "AV_CODEC_ID_ADPCM_G722";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_APC"] = 69661] = "AV_CODEC_ID_ADPCM_IMA_APC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_VIMA"] = 69662] = "AV_CODEC_ID_ADPCM_VIMA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_AFC"] = 69663] = "AV_CODEC_ID_ADPCM_AFC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_OKI"] = 69664] = "AV_CODEC_ID_ADPCM_IMA_OKI";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_DTK"] = 69665] = "AV_CODEC_ID_ADPCM_DTK";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_RAD"] = 69666] = "AV_CODEC_ID_ADPCM_IMA_RAD";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_G726LE"] = 69667] = "AV_CODEC_ID_ADPCM_G726LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_THP_LE"] = 69668] = "AV_CODEC_ID_ADPCM_THP_LE";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_PSX"] = 69669] = "AV_CODEC_ID_ADPCM_PSX";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_AICA"] = 69670] = "AV_CODEC_ID_ADPCM_AICA";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_DAT4"] = 69671] = "AV_CODEC_ID_ADPCM_IMA_DAT4";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_MTAF"] = 69672] = "AV_CODEC_ID_ADPCM_MTAF";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_AGM"] = 69673] = "AV_CODEC_ID_ADPCM_AGM";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_ARGO"] = 69674] = "AV_CODEC_ID_ADPCM_ARGO";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_SSI"] = 69675] = "AV_CODEC_ID_ADPCM_IMA_SSI";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_ZORK"] = 69676] = "AV_CODEC_ID_ADPCM_ZORK";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_APM"] = 69677] = "AV_CODEC_ID_ADPCM_IMA_APM";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_ALP"] = 69678] = "AV_CODEC_ID_ADPCM_IMA_ALP";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_MTF"] = 69679] = "AV_CODEC_ID_ADPCM_IMA_MTF";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_CUNNING"] = 69680] = "AV_CODEC_ID_ADPCM_IMA_CUNNING";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_MOFLEX"] = 69681] = "AV_CODEC_ID_ADPCM_IMA_MOFLEX";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_IMA_ACORN"] = 69682] = "AV_CODEC_ID_ADPCM_IMA_ACORN";
    AVCodecID[AVCodecID["AV_CODEC_ID_ADPCM_XMD"] = 69683] = "AV_CODEC_ID_ADPCM_XMD";
    /* AMR */
    AVCodecID[AVCodecID["AV_CODEC_ID_AMR_NB"] = 73728] = "AV_CODEC_ID_AMR_NB";
    AVCodecID[AVCodecID["AV_CODEC_ID_AMR_WB"] = 73729] = "AV_CODEC_ID_AMR_WB";
    /* RealAudio codecs*/
    AVCodecID[AVCodecID["AV_CODEC_ID_RA_144"] = 77824] = "AV_CODEC_ID_RA_144";
    AVCodecID[AVCodecID["AV_CODEC_ID_RA_288"] = 77825] = "AV_CODEC_ID_RA_288";
    /* various DPCM codecs */
    AVCodecID[AVCodecID["AV_CODEC_ID_ROQ_DPCM"] = 81920] = "AV_CODEC_ID_ROQ_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_INTERPLAY_DPCM"] = 81921] = "AV_CODEC_ID_INTERPLAY_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_XAN_DPCM"] = 81922] = "AV_CODEC_ID_XAN_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_SOL_DPCM"] = 81923] = "AV_CODEC_ID_SOL_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_SDX2_DPCM"] = 81924] = "AV_CODEC_ID_SDX2_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_GREMLIN_DPCM"] = 81925] = "AV_CODEC_ID_GREMLIN_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_DERF_DPCM"] = 81926] = "AV_CODEC_ID_DERF_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_WADY_DPCM"] = 81927] = "AV_CODEC_ID_WADY_DPCM";
    AVCodecID[AVCodecID["AV_CODEC_ID_CBD2_DPCM"] = 81928] = "AV_CODEC_ID_CBD2_DPCM";
    /* audio codecs */
    AVCodecID[AVCodecID["AV_CODEC_ID_MP2"] = 86016] = "AV_CODEC_ID_MP2";
    AVCodecID[AVCodecID["AV_CODEC_ID_MP3"] = 86017] = "AV_CODEC_ID_MP3";
    AVCodecID[AVCodecID["AV_CODEC_ID_AAC"] = 86018] = "AV_CODEC_ID_AAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_AC3"] = 86019] = "AV_CODEC_ID_AC3";
    AVCodecID[AVCodecID["AV_CODEC_ID_DTS"] = 86020] = "AV_CODEC_ID_DTS";
    AVCodecID[AVCodecID["AV_CODEC_ID_VORBIS"] = 86021] = "AV_CODEC_ID_VORBIS";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVAUDIO"] = 86022] = "AV_CODEC_ID_DVAUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMAV1"] = 86023] = "AV_CODEC_ID_WMAV1";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMAV2"] = 86024] = "AV_CODEC_ID_WMAV2";
    AVCodecID[AVCodecID["AV_CODEC_ID_MACE3"] = 86025] = "AV_CODEC_ID_MACE3";
    AVCodecID[AVCodecID["AV_CODEC_ID_MACE6"] = 86026] = "AV_CODEC_ID_MACE6";
    AVCodecID[AVCodecID["AV_CODEC_ID_VMDAUDIO"] = 86027] = "AV_CODEC_ID_VMDAUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_FLAC"] = 86028] = "AV_CODEC_ID_FLAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_MP3ADU"] = 86029] = "AV_CODEC_ID_MP3ADU";
    AVCodecID[AVCodecID["AV_CODEC_ID_MP3ON4"] = 86030] = "AV_CODEC_ID_MP3ON4";
    AVCodecID[AVCodecID["AV_CODEC_ID_SHORTEN"] = 86031] = "AV_CODEC_ID_SHORTEN";
    AVCodecID[AVCodecID["AV_CODEC_ID_ALAC"] = 86032] = "AV_CODEC_ID_ALAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_WESTWOOD_SND1"] = 86033] = "AV_CODEC_ID_WESTWOOD_SND1";
    AVCodecID[AVCodecID["AV_CODEC_ID_GSM"] = 86034] = "AV_CODEC_ID_GSM";
    AVCodecID[AVCodecID["AV_CODEC_ID_QDM2"] = 86035] = "AV_CODEC_ID_QDM2";
    AVCodecID[AVCodecID["AV_CODEC_ID_COOK"] = 86036] = "AV_CODEC_ID_COOK";
    AVCodecID[AVCodecID["AV_CODEC_ID_TRUESPEECH"] = 86037] = "AV_CODEC_ID_TRUESPEECH";
    AVCodecID[AVCodecID["AV_CODEC_ID_TTA"] = 86038] = "AV_CODEC_ID_TTA";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMACKAUDIO"] = 86039] = "AV_CODEC_ID_SMACKAUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_QCELP"] = 86040] = "AV_CODEC_ID_QCELP";
    AVCodecID[AVCodecID["AV_CODEC_ID_WAVPACK"] = 86041] = "AV_CODEC_ID_WAVPACK";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSICINAUDIO"] = 86042] = "AV_CODEC_ID_DSICINAUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_IMC"] = 86043] = "AV_CODEC_ID_IMC";
    AVCodecID[AVCodecID["AV_CODEC_ID_MUSEPACK7"] = 86044] = "AV_CODEC_ID_MUSEPACK7";
    AVCodecID[AVCodecID["AV_CODEC_ID_MLP"] = 86045] = "AV_CODEC_ID_MLP";
    AVCodecID[AVCodecID["AV_CODEC_ID_GSM_MS"] = 86046] = "AV_CODEC_ID_GSM_MS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC3"] = 86047] = "AV_CODEC_ID_ATRAC3";
    AVCodecID[AVCodecID["AV_CODEC_ID_APE"] = 86048] = "AV_CODEC_ID_APE";
    AVCodecID[AVCodecID["AV_CODEC_ID_NELLYMOSER"] = 86049] = "AV_CODEC_ID_NELLYMOSER";
    AVCodecID[AVCodecID["AV_CODEC_ID_MUSEPACK8"] = 86050] = "AV_CODEC_ID_MUSEPACK8";
    AVCodecID[AVCodecID["AV_CODEC_ID_SPEEX"] = 86051] = "AV_CODEC_ID_SPEEX";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMAVOICE"] = 86052] = "AV_CODEC_ID_WMAVOICE";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMAPRO"] = 86053] = "AV_CODEC_ID_WMAPRO";
    AVCodecID[AVCodecID["AV_CODEC_ID_WMALOSSLESS"] = 86054] = "AV_CODEC_ID_WMALOSSLESS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC3P"] = 86055] = "AV_CODEC_ID_ATRAC3P";
    AVCodecID[AVCodecID["AV_CODEC_ID_EAC3"] = 86056] = "AV_CODEC_ID_EAC3";
    AVCodecID[AVCodecID["AV_CODEC_ID_SIPR"] = 86057] = "AV_CODEC_ID_SIPR";
    AVCodecID[AVCodecID["AV_CODEC_ID_MP1"] = 86058] = "AV_CODEC_ID_MP1";
    AVCodecID[AVCodecID["AV_CODEC_ID_TWINVQ"] = 86059] = "AV_CODEC_ID_TWINVQ";
    AVCodecID[AVCodecID["AV_CODEC_ID_TRUEHD"] = 86060] = "AV_CODEC_ID_TRUEHD";
    AVCodecID[AVCodecID["AV_CODEC_ID_MP4ALS"] = 86061] = "AV_CODEC_ID_MP4ALS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC1"] = 86062] = "AV_CODEC_ID_ATRAC1";
    AVCodecID[AVCodecID["AV_CODEC_ID_BINKAUDIO_RDFT"] = 86063] = "AV_CODEC_ID_BINKAUDIO_RDFT";
    AVCodecID[AVCodecID["AV_CODEC_ID_BINKAUDIO_DCT"] = 86064] = "AV_CODEC_ID_BINKAUDIO_DCT";
    AVCodecID[AVCodecID["AV_CODEC_ID_AAC_LATM"] = 86065] = "AV_CODEC_ID_AAC_LATM";
    AVCodecID[AVCodecID["AV_CODEC_ID_QDMC"] = 86066] = "AV_CODEC_ID_QDMC";
    AVCodecID[AVCodecID["AV_CODEC_ID_CELT"] = 86067] = "AV_CODEC_ID_CELT";
    AVCodecID[AVCodecID["AV_CODEC_ID_G723_1"] = 86068] = "AV_CODEC_ID_G723_1";
    AVCodecID[AVCodecID["AV_CODEC_ID_G729"] = 86069] = "AV_CODEC_ID_G729";
    AVCodecID[AVCodecID["AV_CODEC_ID_8SVX_EXP"] = 86070] = "AV_CODEC_ID_8SVX_EXP";
    AVCodecID[AVCodecID["AV_CODEC_ID_8SVX_FIB"] = 86071] = "AV_CODEC_ID_8SVX_FIB";
    AVCodecID[AVCodecID["AV_CODEC_ID_BMV_AUDIO"] = 86072] = "AV_CODEC_ID_BMV_AUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_RALF"] = 86073] = "AV_CODEC_ID_RALF";
    AVCodecID[AVCodecID["AV_CODEC_ID_IAC"] = 86074] = "AV_CODEC_ID_IAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ILBC"] = 86075] = "AV_CODEC_ID_ILBC";
    AVCodecID[AVCodecID["AV_CODEC_ID_OPUS"] = 86076] = "AV_CODEC_ID_OPUS";
    AVCodecID[AVCodecID["AV_CODEC_ID_COMFORT_NOISE"] = 86077] = "AV_CODEC_ID_COMFORT_NOISE";
    AVCodecID[AVCodecID["AV_CODEC_ID_TAK"] = 86078] = "AV_CODEC_ID_TAK";
    AVCodecID[AVCodecID["AV_CODEC_ID_METASOUND"] = 86079] = "AV_CODEC_ID_METASOUND";
    AVCodecID[AVCodecID["AV_CODEC_ID_PAF_AUDIO"] = 86080] = "AV_CODEC_ID_PAF_AUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_ON2AVC"] = 86081] = "AV_CODEC_ID_ON2AVC";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSS_SP"] = 86082] = "AV_CODEC_ID_DSS_SP";
    AVCodecID[AVCodecID["AV_CODEC_ID_CODEC2"] = 86083] = "AV_CODEC_ID_CODEC2";
    AVCodecID[AVCodecID["AV_CODEC_ID_FFWAVESYNTH"] = 86084] = "AV_CODEC_ID_FFWAVESYNTH";
    AVCodecID[AVCodecID["AV_CODEC_ID_SONIC"] = 86085] = "AV_CODEC_ID_SONIC";
    AVCodecID[AVCodecID["AV_CODEC_ID_SONIC_LS"] = 86086] = "AV_CODEC_ID_SONIC_LS";
    AVCodecID[AVCodecID["AV_CODEC_ID_EVRC"] = 86087] = "AV_CODEC_ID_EVRC";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMV"] = 86088] = "AV_CODEC_ID_SMV";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSD_LSBF"] = 86089] = "AV_CODEC_ID_DSD_LSBF";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSD_MSBF"] = 86090] = "AV_CODEC_ID_DSD_MSBF";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSD_LSBF_PLANAR"] = 86091] = "AV_CODEC_ID_DSD_LSBF_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_DSD_MSBF_PLANAR"] = 86092] = "AV_CODEC_ID_DSD_MSBF_PLANAR";
    AVCodecID[AVCodecID["AV_CODEC_ID_4GV"] = 86093] = "AV_CODEC_ID_4GV";
    AVCodecID[AVCodecID["AV_CODEC_ID_INTERPLAY_ACM"] = 86094] = "AV_CODEC_ID_INTERPLAY_ACM";
    AVCodecID[AVCodecID["AV_CODEC_ID_XMA1"] = 86095] = "AV_CODEC_ID_XMA1";
    AVCodecID[AVCodecID["AV_CODEC_ID_XMA2"] = 86096] = "AV_CODEC_ID_XMA2";
    AVCodecID[AVCodecID["AV_CODEC_ID_DST"] = 86097] = "AV_CODEC_ID_DST";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC3AL"] = 86098] = "AV_CODEC_ID_ATRAC3AL";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC3PAL"] = 86099] = "AV_CODEC_ID_ATRAC3PAL";
    AVCodecID[AVCodecID["AV_CODEC_ID_DOLBY_E"] = 86100] = "AV_CODEC_ID_DOLBY_E";
    AVCodecID[AVCodecID["AV_CODEC_ID_APTX"] = 86101] = "AV_CODEC_ID_APTX";
    AVCodecID[AVCodecID["AV_CODEC_ID_APTX_HD"] = 86102] = "AV_CODEC_ID_APTX_HD";
    AVCodecID[AVCodecID["AV_CODEC_ID_SBC"] = 86103] = "AV_CODEC_ID_SBC";
    AVCodecID[AVCodecID["AV_CODEC_ID_ATRAC9"] = 86104] = "AV_CODEC_ID_ATRAC9";
    AVCodecID[AVCodecID["AV_CODEC_ID_HCOM"] = 86105] = "AV_CODEC_ID_HCOM";
    AVCodecID[AVCodecID["AV_CODEC_ID_ACELP_KELVIN"] = 86106] = "AV_CODEC_ID_ACELP_KELVIN";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEGH_3D_AUDIO"] = 86107] = "AV_CODEC_ID_MPEGH_3D_AUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_SIREN"] = 86108] = "AV_CODEC_ID_SIREN";
    AVCodecID[AVCodecID["AV_CODEC_ID_HCA"] = 86109] = "AV_CODEC_ID_HCA";
    AVCodecID[AVCodecID["AV_CODEC_ID_FASTAUDIO"] = 86110] = "AV_CODEC_ID_FASTAUDIO";
    AVCodecID[AVCodecID["AV_CODEC_ID_MSNSIREN"] = 86111] = "AV_CODEC_ID_MSNSIREN";
    AVCodecID[AVCodecID["AV_CODEC_ID_DFPWM"] = 86112] = "AV_CODEC_ID_DFPWM";
    AVCodecID[AVCodecID["AV_CODEC_ID_BONK"] = 86113] = "AV_CODEC_ID_BONK";
    AVCodecID[AVCodecID["AV_CODEC_ID_MISC4"] = 86114] = "AV_CODEC_ID_MISC4";
    AVCodecID[AVCodecID["AV_CODEC_ID_APAC"] = 86115] = "AV_CODEC_ID_APAC";
    AVCodecID[AVCodecID["AV_CODEC_ID_FTR"] = 86116] = "AV_CODEC_ID_FTR";
    AVCodecID[AVCodecID["AV_CODEC_ID_WAVARC"] = 86117] = "AV_CODEC_ID_WAVARC";
    AVCodecID[AVCodecID["AV_CODEC_ID_RKA"] = 86118] = "AV_CODEC_ID_RKA";
    AVCodecID[AVCodecID["AV_CODEC_ID_AC4"] = 86119] = "AV_CODEC_ID_AC4";
    AVCodecID[AVCodecID["AV_CODEC_ID_OSQ"] = 86120] = "AV_CODEC_ID_OSQ";
    AVCodecID[AVCodecID["AV_CODEC_ID_QOA"] = 86121] = "AV_CODEC_ID_QOA";
    /* subtitle codecs */
    AVCodecID[AVCodecID["AV_CODEC_ID_FIRST_SUBTITLE"] = 94208] = "AV_CODEC_ID_FIRST_SUBTITLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVD_SUBTITLE"] = 94208] = "AV_CODEC_ID_DVD_SUBTITLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVB_SUBTITLE"] = 94209] = "AV_CODEC_ID_DVB_SUBTITLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_TEXT"] = 94210] = "AV_CODEC_ID_TEXT";
    AVCodecID[AVCodecID["AV_CODEC_ID_XSUB"] = 94211] = "AV_CODEC_ID_XSUB";
    AVCodecID[AVCodecID["AV_CODEC_ID_SSA"] = 94212] = "AV_CODEC_ID_SSA";
    AVCodecID[AVCodecID["AV_CODEC_ID_MOV_TEXT"] = 94213] = "AV_CODEC_ID_MOV_TEXT";
    AVCodecID[AVCodecID["AV_CODEC_ID_HDMV_PGS_SUBTITLE"] = 94214] = "AV_CODEC_ID_HDMV_PGS_SUBTITLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVB_TELETEXT"] = 94215] = "AV_CODEC_ID_DVB_TELETEXT";
    AVCodecID[AVCodecID["AV_CODEC_ID_SRT"] = 94216] = "AV_CODEC_ID_SRT";
    AVCodecID[AVCodecID["AV_CODEC_ID_MICRODVD"] = 94217] = "AV_CODEC_ID_MICRODVD";
    AVCodecID[AVCodecID["AV_CODEC_ID_EIA_608"] = 94218] = "AV_CODEC_ID_EIA_608";
    AVCodecID[AVCodecID["AV_CODEC_ID_JACOSUB"] = 94219] = "AV_CODEC_ID_JACOSUB";
    AVCodecID[AVCodecID["AV_CODEC_ID_SAMI"] = 94220] = "AV_CODEC_ID_SAMI";
    AVCodecID[AVCodecID["AV_CODEC_ID_REALTEXT"] = 94221] = "AV_CODEC_ID_REALTEXT";
    AVCodecID[AVCodecID["AV_CODEC_ID_STL"] = 94222] = "AV_CODEC_ID_STL";
    AVCodecID[AVCodecID["AV_CODEC_ID_SUBVIEWER1"] = 94223] = "AV_CODEC_ID_SUBVIEWER1";
    AVCodecID[AVCodecID["AV_CODEC_ID_SUBVIEWER"] = 94224] = "AV_CODEC_ID_SUBVIEWER";
    AVCodecID[AVCodecID["AV_CODEC_ID_SUBRIP"] = 94225] = "AV_CODEC_ID_SUBRIP";
    AVCodecID[AVCodecID["AV_CODEC_ID_WEBVTT"] = 94226] = "AV_CODEC_ID_WEBVTT";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPL2"] = 94227] = "AV_CODEC_ID_MPL2";
    AVCodecID[AVCodecID["AV_CODEC_ID_VPLAYER"] = 94228] = "AV_CODEC_ID_VPLAYER";
    AVCodecID[AVCodecID["AV_CODEC_ID_PJS"] = 94229] = "AV_CODEC_ID_PJS";
    AVCodecID[AVCodecID["AV_CODEC_ID_ASS"] = 94230] = "AV_CODEC_ID_ASS";
    AVCodecID[AVCodecID["AV_CODEC_ID_HDMV_TEXT_SUBTITLE"] = 94231] = "AV_CODEC_ID_HDMV_TEXT_SUBTITLE";
    AVCodecID[AVCodecID["AV_CODEC_ID_TTML"] = 94232] = "AV_CODEC_ID_TTML";
    AVCodecID[AVCodecID["AV_CODEC_ID_ARIB_CAPTION"] = 94233] = "AV_CODEC_ID_ARIB_CAPTION";
    /* other specific kind of codecs (generally used for attachments) */
    AVCodecID[AVCodecID["AV_CODEC_ID_FIRST_UNKNOWN"] = 98304] = "AV_CODEC_ID_FIRST_UNKNOWN";
    AVCodecID[AVCodecID["AV_CODEC_ID_TTF"] = 98304] = "AV_CODEC_ID_TTF";
    AVCodecID[AVCodecID["AV_CODEC_ID_SCTE_35"] = 98305] = "AV_CODEC_ID_SCTE_35";
    AVCodecID[AVCodecID["AV_CODEC_ID_EPG"] = 98306] = "AV_CODEC_ID_EPG";
    AVCodecID[AVCodecID["AV_CODEC_ID_BINTEXT"] = 98307] = "AV_CODEC_ID_BINTEXT";
    AVCodecID[AVCodecID["AV_CODEC_ID_XBIN"] = 98308] = "AV_CODEC_ID_XBIN";
    AVCodecID[AVCodecID["AV_CODEC_ID_IDF"] = 98309] = "AV_CODEC_ID_IDF";
    AVCodecID[AVCodecID["AV_CODEC_ID_OTF"] = 98310] = "AV_CODEC_ID_OTF";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMPTE_KLV"] = 98311] = "AV_CODEC_ID_SMPTE_KLV";
    AVCodecID[AVCodecID["AV_CODEC_ID_DVD_NAV"] = 98312] = "AV_CODEC_ID_DVD_NAV";
    AVCodecID[AVCodecID["AV_CODEC_ID_TIMED_ID3"] = 98313] = "AV_CODEC_ID_TIMED_ID3";
    AVCodecID[AVCodecID["AV_CODEC_ID_BIN_DATA"] = 98314] = "AV_CODEC_ID_BIN_DATA";
    AVCodecID[AVCodecID["AV_CODEC_ID_SMPTE_2038"] = 98315] = "AV_CODEC_ID_SMPTE_2038";
    AVCodecID[AVCodecID["AV_CODEC_ID_PROBE"] = 102400] = "AV_CODEC_ID_PROBE";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEG2TS"] = 131072] = "AV_CODEC_ID_MPEG2TS";
    AVCodecID[AVCodecID["AV_CODEC_ID_MPEG4SYSTEMS"] = 131073] = "AV_CODEC_ID_MPEG4SYSTEMS";
    AVCodecID[AVCodecID["AV_CODEC_ID_FFMETADATA"] = 135168] = "AV_CODEC_ID_FFMETADATA";
    AVCodecID[AVCodecID["AV_CODEC_ID_WRAPPED_AVFRAME"] = 135169] = "AV_CODEC_ID_WRAPPED_AVFRAME";
    /**
     * Dummy null video codec, useful mainly for development and debugging.
     * Null encoder/decoder discard all input and never return any output.
     */
    AVCodecID[AVCodecID["AV_CODEC_ID_VNULL"] = 135170] = "AV_CODEC_ID_VNULL";
    /**
     * Dummy null audio codec, useful mainly for development and debugging.
     * Null encoder/decoder discard all input and never return any output.
     */
    AVCodecID[AVCodecID["AV_CODEC_ID_ANULL"] = 135171] = "AV_CODEC_ID_ANULL";
})(AVCodecID || (AVCodecID = {}));
;
